extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageConfig.java

```diff
@@ -61,4 +61,20 @@
      */
     @ConfigItem
     public boolean push;
+
+    /**
+     * Since user.name which is default value can be uppercase and uppercase values are not allowed
+     * in the repository part of image references, we need to make the username lowercase.
+     *
+     * We purposely don't change the value of an explicitly set group.
+     */
+    public Optional<String> getEffectiveGroup() {
+        if (group.isPresent()) {
+            String originalGroup = group.get();
+            if (originalGroup.equals(System.getProperty("user.name"))) {
+                return Optional.of(originalGroup.toLowerCase());
```

Comment:
I'm not sure why we lowercase only in this case? I mean it will fail if the group has some uppercase character, won't it?

Reply:
My idea is that it's better to fail if the user explicitly set a group on their own that is invalid (the error message is very explicit and actionable). I am thinking that if we start changing user supplied configuration it could be a very slippery slope...
If however the user did not set anything and the username was used, then onlt in that implicit case should intervene.